array_api_compat/dask/array/_aliases.py

```diff
@@ -39,7 +39,14 @@
 
 isdtype = get_xp(np)(_aliases.isdtype)
 unstack = get_xp(da)(_aliases.unstack)
-astype = _aliases.astype
+
+def astype(x: Array, dtype: Dtype, /, *, copy: bool = True) -> Array:
+    if not copy and dtype == x.dtype:
+        return x
+    # dask astype doesn't respect copy=True so copy
+    # manually via numpy
+    x = np.array(x, dtype=dtype, copy=copy)
```

Comment:
Hm, thinking about this again, maybe I should do something like
```
x = x.astype(dtype=dtype)
if copy:
    x = x.map_blocks(lambda x: x.copy())
return x
```
to prevent the graph break?